tests/integration/testproofs.nim

```diff
@@ -62,12 +62,17 @@ twonodessuite "Proving integration test", debug1=false, debug2=false:
   proc startValidator: NodeProcess =
     let validator = startNode(
       [
+        "persistence",
+        "validator",
         "--data-dir=" & validatorDir,
         "--api-port=8089",
         "--disc-port=8099",
         "--listen-addrs=/ip4/127.0.0.1/tcp/0",
-        "--validator",
-        "--eth-account=" & $accounts[2]
+        "--eth-account=" & $accounts[2],
+        # TODO: this is wrong, validator shouldn't need circom stuff
```

Comment:
we can move validator to be it's own command it this isn't needed for it

Reply:
can you clarify what parameters are needed for the validator?

Reply:
The validator only needs to interact with the smart contract, so these parameters would apply:
--eth-provider
--eth-account
--eth-private-key
--marketplace-address

Reply:
OK, these are shared parameters between the prover and verifier, I can add a flag for `circuits`, which would then make it mandatory to either select either a `persistence & prover` or `persistence & validator`.

Reply:
> I can add a flag for `circuits`, which would then make it mandatory to either select either a `persistence & prover` or `persistence & validator`

I don't understand this? Keep in mind that the validator does not check zk proofs, that is the job of the smart contract. The validator only looks for missed proofs, and points them out to the smart contract. Only a host (prover) needs the circuit parameters, because it needs to create proofs.

Reply:
```sh
Available sub-commands:

codex persistence [OPTIONS]... command

The following options are available:

     --eth-provider         The URL of the JSON-RPC API of the Ethereum node [=ws://localhost:8545].
     --eth-account          The Ethereum account that is used for storage contracts.
     --eth-private-key      File containing Ethereum private key for storage contracts.
     --marketplace-address  Address of deployed Marketplace contract.

Available sub-commands:

codex persistence prover [OPTIONS]...

The following options are available:

     --circom-r1cs          The r1cs file for the storage circuit [=/Users/dryajov/Library/Application
                            Support/Codex/circuits/proof_main.r1cs].
     --circom-wasm          The wasm file for the storage circuit [=/Users/dryajov/Library/Application
                            Support/Codex/circuits/proof_main.wasm].
     --circom-zkey          The zkey file for the storage circuit [=/Users/dryajov/Library/Application
                            Support/Codex/circuits/proof_main.zkey].
     --circom-no-zkey       Ignore the zkey file - use only for testing! [=false].
     --proof-samples        Number of samples to prove [=5].
     --max-slot-depth       The maximum depth of the slot tree [=32].
     --max-dataset-depth    The maximum depth of the dataset tree [=8].
     --max-block-depth      The maximum depth of the network block merkle tree [=5].
     --max-cell-elements    The maximum number of elements in a cell [=67].

codex persistence validator [OPTIONS]...

The following options are available:

     --validator-max-slots  Maximum number of slots that the validator monitors [=1000].
```

Reply:
Right, that makes sense.
The we'd also have to check that the `persistence prover` options are set when starting the sales module.

Reply:
However, this will only allow you to be a prover or a validator, not both. Is there a case where you can be both? In which case, we can just leave `--validator` as cmd argument rather than a command.

Reply:
> However, this will only allow you to be a prover or a validator, not both. Is there a case where you can be both? In which case, we can just leave `--validator` as cmd argument rather than a command.

I don't think there is anything limiting a node from being both a validator and a host (prover). A host should be able to mark proofs as missing (including its own).

Reply:
yep, thats what we agreed on, we should have `--validator` as a cmd param.